pom.xml

```diff
@@ -217,6 +220,8 @@
               <exclude>NOTICE-*</exclude>
               <exclude>src/main/java/org/terracotta/org/junit/**</exclude>
               <exclude>src/test/java/org/terracotta/org/junit/**</exclude>
+              <exclude>.github/**</exclude>
+              <exclude>**/.m2/**</exclude>
```

Comment:
Why is this (`.m2`) necessary?

Reply:
Because when running in docker in user mode, the .m2 is created inside the project folder. I think this is because the home folder of 1001 is the project folder

Reply:
This strikes me somewhat as the tail wagging the dog (if I can use a colloquialism).  Can we not set up the container environment to avoid this?

Reply:
Personally it does not strike me because I am used to that, especially when running build in the cloud on external service providers that are cheap and free so they provide only containers with limited access.

@akomakom might know better but from my experience this is common that when running as a user the home folder is set to be where the project is checkout but maybe we can control that somehow... I just don't know. 

And this is not so different than the maven wrapper plugin that is downloading things within the project and that we could ignore...

Reply:
It's necessary for only one variant of a container-based build.  It's more appropriate to alter the container environment, if possible, than to have pom content that a local build user will be clueless about.

Reply:
Will check tomorrow. I agree but really don't know how it could be done other than controlling where the checkout happens because I don't know if we can control the home folder to not be the checkout location when running as user 1001...

Reply:
The reason your home dir is the workspace is because `--user 1001` is a hack.  It doesn't actually add a user to the container's `/etc/passwd`, hence there is no home directory.   The proper way is of course to build a custom image with that user propely added via `useradd`.

One workaround is to specify `MAVEN_HOME` or `-Dmaven.repo.local`

Reply:
> The reason your home dir is the workspace is because `--user 1001` is a hack. It doesn't actually add a user to the container's `/etc/passwd`, hence there is no home directory, and the CWD is `/`. The proper way is of course to build a custom image with that user propely added via `useradd`.
> 
> One workaround is to specify `MAVEN_HOME` or `-Dmaven.repo.local`

Yes, I thought about fixing it like that at first, but this particular maven parameter doesn't solve globally the problem of the user.home system prop seen as ? in the jvm, so in the maven wrapper code and test jvm. It would only work if nothing within the project is using user.home which is not a warranty now or in the future.

Ensuring user.home is set is IMO fixing the source of the symptoms, not only 1 symptom.